tests/rest/client/test_shadow_banned.py

```diff
@@ -176,3 +178,34 @@ def test_upgrade(self):
         summary = self.get_success(self.store.get_room_summary(new_room_id))
         # The summary should be empty since the room doesn't exist.
         self.assertEqual(summary, {})
+
+
+# To avoid the tests timing out don't add a delay to "annoy the requester".
+@patch("random.randint", new=lambda a, b: 0)
+class ProfileTestCase(_ShadowBannedBase):
+    servlets = [
+        synapse.rest.admin.register_servlets_for_client_rest_resource,
+        login.register_servlets,
+        profile.register_servlets,
+    ]
+
+    def test_displayname(self):
+        """Profile changes should succeed, but don't end up in a room."""
+        # The update should succeed.
+        request, channel = self.make_request(
+            "PUT",
+            "/_matrix/client/r0/profile/%s/displayname" % (self.banned_user_id,),
+            {"displayname": "new display name"},
+            access_token=self.banned_access_token,
+        )
+        self.render(request)
+        self.assertEquals(200, channel.code, channel.result)
+        self.assertEqual(channel.json_body, {})
+
+        # The user's display name should be updated.
+        request, channel = self.make_request(
+            "GET", "/profile/%s/displayname" % (self.banned_user_id,)
+        )
+        self.render(request)
+        self.assertEqual(channel.code, 200, channel.result)
+        self.assertEqual(channel.json_body["displayname"], "new display name")
```

Comment:
Can you stick the user in a room and make sure their `m.room.member` event still contains the old displayname?

Reply:
Yes! I wasn't sure how to assert that. Thank you! 👍

Reply:
Yep, no need for syncing or anything, just use [the GET equivalent](https://matrix.org/docs/spec/client_server/unstable#get-matrix-client-r0-rooms-roomid-state-eventtype-statekey).